polaris-react/src/components/TextField/TextField.tsx

```diff
@@ -656,16 +660,23 @@ export function TextField({
 
   function handleKeyPress(event: React.KeyboardEvent) {
     const {key, which} = event;
-    const numbersSpec = /[\d.eE+-]$/;
-    if (type !== 'number' || which === Key.Enter || numbersSpec.test(key)) {
+    const numbersSpec = /[\d.,eE+-]$/;
```

Comment:
Suggestion:
Add the comma `,` to the number spec so that users in regions where the comma is used as the decimal separator ([Europe, Africa, South America](https://en.wikipedia.org/wiki/Decimal_separator#/media/File:DecimalSeparator.svg)) can use the correct key. Currently, all users, regardless of location, must use the period `.` to indicate a decimal place.
Handling of the decimal separator is done by the html `input` component based on the browser's language. The code will always receive the decimal point as a period `.` regardless of which separator the user entered, so we do not need to make any additional allowances to accommodate this.
The result is that a user in North America will be able to use the decimal separator `.`, whilst a user in Europe can use either `.` or `,`.

Additionally, there is one edge case in Safari to be aware of. Safari's number type input boxes do not appear to limit the input to one decimal character (i.e. `123.4.5.6.7.8` is a valid input). Currently, Shopify does not validate to prevent the user from typing in extra decimal points. Combined with this regex change, users in Europe may start to think they can enter thousand separators into their numbers. My recommendation is to add a regex to prevent more than one `.` or `,` being entered into the input field, but I'm interested in ideas/thoughts on this.